sdc_client/client.py

```diff
@@ -35,12 +35,21 @@ def _get_client(streamsets: IStreamSets) -> _StreamSetsApiClient:
     return _clients[streamsets.get_id()]
 
 
+def _get_streamsets_for_pipeline(pipeline: IPipeline):
+    streamsets_pipelines = balancer.get_streamsets_pipelines()
+    streamsets_with_preferred_type = list(filter(lambda ss: ss.get_preferred_type() == pipeline.source_type, streamsets_pipelines))
+    if streamsets_with_preferred_type:
+        return balancer.least_loaded_streamsets({
+            streamsets_: streamsets_pipelines[streamsets_] for streamsets_ in streamsets_with_preferred_type
+        })
```

Comment:
I'm not sure if I right understand. Here is filtering `streamsets_pipelines` and then inside if we execute kind of filtering again. Could we combine these two and return `streamsets_with_preferred_type` as a desirable dict?